### PR TITLE
[Android] Filtered Input.ChoiceSet implementation

### LINKED ISSUE #5984: [Android] Input.ChoiceSet static filtering

### PR DESCRIPTION
**Pull request by almedina-ms**
# Related Issue

Fixes #5984 

# Description

Implements the `Filtered` style for `Input.ChoiceSet` which allows the user to type in to get filtered results inside of a dropdown. This code includes the implementation of the control as well as the inclusion of tests to verify the correct validation of the data in the control. 

# Sample Card

v1.5\Elements\Input.ChoiceSet.Filtered.json

# How Verified

The fix was verified by adding tests to verify:
1. Serialization and Deserialization of the property 
2. Validation success and failure cases

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6009)

### CODE REVIEW

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/input/ChoiceSetInputRenderer.java [hunk 1]
```diff
@@ -384,6 +387,121 @@ public void onNothingSelected(AdapterView<?> parent)
...
+            autoCompleteTextInputHandler.setView(inputLayout);
+        }
+        else
+        {
+            autoCompleteTextView.setTag(new TagContent(choiceSetInput, autoCompleteTextInputHandler));
+            autoCompleteTextInputHandler.setView(autoCompleteTextView);
+        }
+        renderedCard.registerInputHandler(autoCompleteTextInputHandler, renderArgs.getContainerCardId());
+
+        class FilteredChoiceSetAdapter extends ArrayAdapter<String>
```
Comment (jwoo-msft) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/input/ChoiceSetInputRenderer.java:439]: nit: can you add a brief description of what this class does?

**Review by jwoo-msft — APPROVED** [review]
would appreciate it if you can add a few screenshots.

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/input/customcontrols/ValidatedAutoCompleteTextView.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+import androidx.core.content.ContextCompat;
+
+import io.adaptivecards.R;
+
+/**
+ * ValidatedAutoCompleteTextView is an extension of AutoCompleteTextView that supports the usage of
+ * the adaptive_state_input_invalid boolean flag defined in attrs.xml for using a custom background drawable
+ */
+@SuppressLint("AppCompatCustomView")
```
Comment (golddove) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/input/customcontrols/ValidatedAutoCompleteTextView.java:18]: It seems to be recommended to extend [`AppCompatAutocompleteTextView`](https://developer.android.com/reference/androidx/appcompat/widget/AppCompatAutoCompleteTextView) for back-compat, which might fix this warning

**Review by golddove — CHANGES_REQUESTED** [review]

**Review by golddove — APPROVED** [review]

**Pull request merged**